File: CS/WindowsFormsApplication18/MRUComboBoxEditPopupListBoxPainter.cs
using System.Drawing;
using DevExpress.Utils.Drawing.Helpers;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace MRUComboBox
{
	public class MRUComboBoxEditPopupListBoxPainter : BaseListBoxPainter
	{
		public MRUComboBoxEditPopupListBoxPainter()
		{
		}

		protected override void DrawItemCore(ControlGraphicsInfoArgs info, BaseListBoxViewInfo.ItemInfo itemInfo, ListBoxDrawItemEventArgs e)
		{
			base.DrawItemCore(info, itemInfo, e);

			MRUComboBoxEditPopupListBoxViewInfo viewInfo = info.ViewInfo as MRUComboBoxEditPopupListBoxViewInfo;
			if ( viewInfo == null )
				return;

			MRUComboBoxEditPopupListBox popupListBox = viewInfo.Owner as MRUComboBoxEditPopupListBox;
			if ( popupListBox == null )
				return;

			MRUComboBoxEdit comboBox = popupListBox.OwnerEdit as MRUComboBoxEdit;
			if ( comboBox == null )
				return;

			if ( itemInfo.Index == comboBox.Properties.LineItemIndex )
			{
				Color borderColor = NativeVista.IsVista ? SystemColors.ControlDark : SystemColors.InactiveBorder;

				using ( Pen pen = new Pen(borderColor) )
				{
					e.Graphics.DrawLine(pen, new Point(e.Bounds.Left, e.Bounds.Bottom - 2), new Point(e.Bounds.Right, e.Bounds.Bottom - 2));
					e.Graphics.DrawLine(pen, new Point(e.Bounds.Left, e.Bounds.Bottom), new Point(e.Bounds.Right, e.Bounds.Bottom));
				}
			}
		}
	}
}

File: VB/WindowsFormsApplication18/MRUComboBoxEditPopupListBoxPainter.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports DevExpress.Utils.Drawing.Helpers
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace MRUComboBox
	Public Class MRUComboBoxEditPopupListBoxPainter
		Inherits BaseListBoxPainter
		Public Sub New()
		End Sub

		Protected Overrides Sub DrawItemCore(ByVal info As ControlGraphicsInfoArgs, ByVal itemInfo As BaseListBoxViewInfo.ItemInfo, ByVal e As ListBoxDrawItemEventArgs)
			MyBase.DrawItemCore(info, itemInfo, e)

			Dim viewInfo As MRUComboBoxEditPopupListBoxViewInfo = TryCast(info.ViewInfo, MRUComboBoxEditPopupListBoxViewInfo)
			If viewInfo Is Nothing Then
				Return
			End If

			Dim popupListBox As MRUComboBoxEditPopupListBox = TryCast(viewInfo.Owner, MRUComboBoxEditPopupListBox)
			If popupListBox Is Nothing Then
				Return
			End If

			Dim comboBox As MRUComboBoxEdit = TryCast(popupListBox.OwnerEdit, MRUComboBoxEdit)
			If comboBox Is Nothing Then
				Return
			End If

			If itemInfo.Index = comboBox.Properties.LineItemIndex Then
				Dim borderColor As Color = If(NativeVista.IsVista, SystemColors.ControlDark, SystemColors.InactiveBorder)

				Using pen As New Pen(borderColor)
					e.Graphics.DrawLine(pen, New Point(e.Bounds.Left, e.Bounds.Bottom - 2), New Point(e.Bounds.Right, e.Bounds.Bottom - 2))
					e.Graphics.DrawLine(pen, New Point(e.Bounds.Left, e.Bounds.Bottom), New Point(e.Bounds.Right, e.Bounds.Bottom))
				End Using
			End If
		End Sub
	End Class
End Namespace
